File: Java/Assignment4/Assignment4_5.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// file name   : Assignment4_5.java
// Description : write a program which accept number form user and return differnce  of  its factors and  non factors sum .
// Author      : Aniket Devaba Markad                        
// Date        : 6/12/2025 
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Number
{
    private int iNo;

    Number(int a)
    {
        this.iNo = a;

        if(iNo < 0)
    {
        iNo = -iNo;
    }
    
    }
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name : SumDiffrance
// Description   : This function is used to calculate diffrance between  sum of factors and  Non factors . 
// Author        : Aniket Devaba Markad 
// Date          : 6/12/2025 
// 
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    public int SumDiffrance()
    {
        int iCnt = 0,iSum1 = 0,iSum2 = 0 ,iDiff = 0;

        for(iCnt = iNo; iCnt >= 1;iCnt --)
        {
            if(iNo % iCnt != 0)
            {
                iSum1 = iSum1 + iCnt;
            }
            else
            {
                iSum2 = iSum2 + iCnt;
            }
        }
            iDiff = iSum2 - iSum1;
        
        return iDiff;
    }

}

public class Assignment4_5
{
    public static void main (String a[])
    {
        int iValue = 0,iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("enter the number : ");
        iValue = sobj.nextInt();

        Number nobj = new Number(iValue);

        iRet = nobj.SumDiffrance();
         
        if(iRet < 0)
        {
            iRet = -(iRet);
        }
        System.out.println(" Diffrance of factors and nonfactors is : "+iRet);
    }
}

File: CPP/Assignment4_5.cpp
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//  File name   : Assignment4_5.cpp
//  Description : Write a program which accept number from user and return difference between factors & non factors.
//  Author      : Aniket Devaba Markad
//  Date        : 10/12/2025
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include <iostream>
using namespace std;

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//  Function Name : FactorDiff
//  Description   : This function is used to calculate difference between factors and non factors.
//  Input         : int
//  Output        : int
//  Author        : Aniket Devaba Markad
//  Date          : 10/12/2025
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

int FactorDiff(int iNo)
{
    int iCnt = 0;
    int iSum1 = 0;
    int iSum2 = 0;
    int iDiff = 0;

    if(iNo < 0)
    {
        iNo = -iNo;
    }

    for(iCnt = 1; iCnt < iNo; iCnt++)
    {
        if((iNo % iCnt) != 0)
        {
            iSum1 = iSum1 + iCnt;
        }
        else
        {
            iSum2 = iSum2 + iCnt;
        }
    }

    iDiff = iSum1 - iSum2;
    return iDiff;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout << "Enter number: ";
    cin >> iValue;

    iRet = FactorDiff(iValue);

    cout << "Difference is: " << iRet << endl;

    return 0;
}

// Time Complexity : O(N)
